src/components/mainpage/sponsors/sponsors.js

```diff
@@ -0,0 +1,36 @@
+import React from "react"
+import sponsorsStyles from "../../../styles/sponsors.module.css"
+import SponsorSection, { getSponsors } from "../sponsors/sponsor-section"
+import { useSponsors } from "../../hooks/sponsors-query"
+
+const Sponsors = () => {
+  const data = useSponsors()
+  const mainSponsors = getSponsors(data, "main")
```

Comment:
These data queries are only being used for checking the length but you could pass them to the sponsors section instead of querying again to get the data there.